lib/plugins/rest.js

```diff
@@ -161,6 +161,40 @@ exports.register = function (server, options, next) {
 			});
 		}
 	});
+
+	server.route({
+		method  : "PUT",
+		path    : "/v1/instances/{instanceId}",
+		handler : function (request, reply) {
+			instances.getInstance({ id : request.params.instanceId })
+			.then(function (response) {
+
+				var instance = new Instance({
+					id    : request.params.instanceId,
+					ami   : request.payload.ami,
+					type  : request.payload.type,
+					state : request.payload.state,
+					uri   : request.payload.uri
+				});
+				var Constructor = instance.constructor;
```

Comment:
This seems unnecessary since we explicitly used the `Instance` constructor in the previous expression. Could you explain why this was needed?